### PR TITLE
feature: MySQL DDL Statements

### PR DESCRIPTION
**Pull request by iobruno**
## Summary

- Add MySQL DDLs (File: `mysql.ddl`)
  - Changes table name from `groups` to `artifactgroups` for MySQL, since it's a registered keyword
  - Add `GetNextSequenceValue` store procedure since MySQL doesn't have an equivalent to Postgres' RETURNING or SQLServer's OUTPUT
  - Dropped IDX_versions_4 on versions as MySQL can't index VARCHAR(1024)
  - Dropped IDX_props_2 on properties as MySQL can't index VARCHAR(1024)
- Configured and fixed several testContainers for MySQL
  - Fix pattern to identify PK violation on MySQL
- Overlay config for storage/mysql
- Registers "mysql" jdbc driver for eligiblity for nextSequenceValue on AbstractSqlRegistryStorage
- Add and Registers MySqlRegistryStorage for Maven

Note*⚠️ A few compromises had to be made to make it compatible with MySQL:

- On artifacts table:
  - Reduced groupId and artifactId to VARCHAR(256) from VARCHAR(512) due to the Primary Key exceeding the MySQL limit of 3072 bytes
- On rules table:
  - Reduced groupId and artifactId to VARCHAR(256) from VARCHAR(512) due to the Primary Key exceeding the MySQL limit of 3072 bytes
- On versions table:
  - Reduced groupId and artifactId to VARCHAR(256) from VARCHAR(512) due to the Primary Key exceeding the MySQL limit of 3072 bytes
  - Reduced version to VARCHAR(128) from VARCHAR(256) due to the constraint UQ_versions_1 exceeding the MySQL limit of 3072 bytes
  - Reduced description to VARCHAR(512) from VARCHAR(1024) due to the IDX_versions_4 exceeding the MySQL limit of 3072 bytes
- On properties table:
  - Reduced pvalue to VARCHAR(512) from VARCHAR(1024) due to the IDX_props_2 exceeding the MySQL limit of 3072 bytes

### CODE REVIEW

File: storage/mysql/src/test/java/io/apicurio/registry/storage/util/MySqlEmbeddedTestResource.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    @Override
+    public Map<String, String> start() {
+        System.out.println("[MySqlEmbeddedTestResource] MySQL test resource starting.");
+
+        String currentEnv = System.getenv("CURRENT_ENV");
+
+        if ("mas".equals(currentEnv)) {
+            Map<String, String> props = new HashMap<>();
+            props.put("quarkus.datasource.jdbc.url", "jdbc:sqlserver://mysql;");
```
Comment (gr8routdoors) [inline storage/mysql/src/test/java/io/apicurio/registry/storage/util/MySqlEmbeddedTestResource.java]: This is incorrect

File: storage/mysql/src/test/java/io/apicurio/registry/storage/util/MySqlEmbeddedTestResource.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    /**
+     * @see io.quarkus.test.common.QuarkusTestResourceLifecycleManager#stop()
+     */
+    @Override
+    public void stop() {
+        if (database != null) {
+            database.close();
+        }
+        System.out.println("[MsSqlEmbeddedTestResource] MS SQLServer test resource stopped.");
```
Comment (gr8routdoors) [inline storage/mysql/src/test/java/io/apicurio/registry/storage/util/MySqlEmbeddedTestResource.java]: This should be changed

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/MySQLSqlStatements.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                "VALUES (?, ?)",
+                "ON DUPLICATE KEY UPDATE loglevel = ?;");
+    }
+
+    /**
+     * @see SqlStatements#getNextSequenceValue()
+     */
+    @Override
+    public String getNextSequenceValue() {
+        //TODO: Create a Stored Procedure to return the VALUE after the UPSERT
```
Comment (gr8routdoors) [inline app/src/main/java/io/apicurio/registry/storage/impl/sql/MySQLSqlStatements.java]: Are you sure you need a stored procedure here?  Seems overkill.

File: storage/mysql/src/test/java/io/apicurio/registry/storage/util/MySqlEmbeddedTestResource.java (hunk 1 above)
Comment (iobruno) [reply storage/mysql/src/test/java/io/apicurio/registry/storage/util/MySqlEmbeddedTestResource.java]: Missed the last push

File: storage/mysql/src/test/java/io/apicurio/registry/storage/util/MySqlEmbeddedTestResource.java (hunk 2 above)
Comment (iobruno) [reply storage/mysql/src/test/java/io/apicurio/registry/storage/util/MySqlEmbeddedTestResource.java]: Forgot to commit the last changes on this. But it's fixed now.

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/MySQLSqlStatements.java (hunk 3 above)
Comment (iobruno) [reply app/src/main/java/io/apicurio/registry/storage/impl/sql/MySQLSqlStatements.java]: I've got to test this some more, I took the MySQL and SQLServer implementation as the basis and MySQL doesn't have the equivalent. For instance:

**Postgres**:
```
    @Override
    public String getNextSequenceValue() {
        return "INSERT INTO sequences (tenantId, name, value) VALUES (?, ?, 1) ON CONFLICT (tenantId, name) DO UPDATE SET value = sequences.value + 1 RETURNING value";
    }
```  

**SQLServer**:
```
    @Override
    public String getNextSequenceValue() {
        return String.join(" ",
                "MERGE INTO sequences AS target",
                "USING (VALUES  (?, ?)) AS source (tenantId, name)",
                "ON (target.tenantId = source.tenantId AND target.name = source.name)",
                "WHEN MATCHED THEN",
                    "UPDATE SET value = target.value + 1",
                "WHEN NOT MATCHED THEN",
                    "INSERT (tenantId, name, value)",
                    "VALUES (source.tenantId, source.name, 1)",
                "OUTPUT INSERTED.value;");
    }
```

**Pull request closed without merging**